sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueClientBuilder.java

```diff
@@ -193,7 +193,16 @@ public QueueClientBuilder() {
      * and {@link #retryOptions(RequestRetryOptions)} have been set.
      */
     public QueueClient buildClient() {
-        return new QueueClient(buildAsyncClient());
+        StorageImplUtils.assertNotNull("queueName", queueName);
+        if (processMessageDecodingErrorAsyncHandler != null && processMessageDecodingErrorHandler != null) {
```

Comment:
Also, can both sync and async error handlers be null? Should we make one of them required depending on what type of client is being created?